hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/ReplicationManagerMetrics.java

```diff
@@ -0,0 +1,161 @@
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.hadoop.hdds.scm.container.replication;
+
+import org.apache.hadoop.hdds.scm.container.ReplicationManager;
+import org.apache.hadoop.metrics2.annotation.Metric;
+import org.apache.hadoop.metrics2.annotation.Metrics;
+import org.apache.hadoop.metrics2.lib.DefaultMetricsSystem;
+import org.apache.hadoop.metrics2.lib.MutableCounterLong;
+import org.apache.hadoop.metrics2.lib.MutableGaugeLong;
+import org.apache.hadoop.ozone.OzoneConsts;
+
+/**
+ * Class contains metrics related to ReplicationManager.
+ */
+@Metrics(about = "Replication Manager Metrics", context = OzoneConsts.OZONE)
+public final class ReplicationManagerMetrics {
+
+  public static final String METRICS_SOURCE_NAME =
+      ReplicationManagerMetrics.class.getSimpleName();
+
+  @Metric("Tracked inflight container replication requests.")
+  private MutableGaugeLong inflightReplication;
+
+  @Metric("Tracked inflight container deletion requests.")
+  private MutableGaugeLong inflightDeletion;
+
+  @Metric("Tracked inflight container move requests.")
+  private MutableGaugeLong inflightMove;
+
+  @Metric("Number of replication commands sent.")
+  private MutableCounterLong numReplicationCmdsSent;
+
+  @Metric("Number of replication commands completed.")
+  private MutableCounterLong numReplicationCmdsCompleted;
+
+  @Metric("Number of replication commands timeout.")
+  private MutableCounterLong numReplicationCmdsTimeout;
+
+  @Metric("Number of deletion commands sent.")
+  private MutableCounterLong numDeletionCmdsSent;
+
+  @Metric("Number of deletion commands completed.")
+  private MutableCounterLong numDeletionCmdsCompleted;
+
+  @Metric("Number of deletion commands timeout.")
+  private MutableCounterLong numDeletionCmdsTimeout;
+
+  @Metric("Number of replication bytes total.")
+  private MutableCounterLong numReplicationBytesTotal;
+
+  @Metric("Number of replication bytes completed.")
+  private MutableCounterLong numReplicationBytesCompleted;
+
+  private ReplicationManager replicationManager;
+
+  public ReplicationManagerMetrics(ReplicationManager manager) {
+    this.replicationManager = manager;
+  }
+
+  public static ReplicationManagerMetrics create(ReplicationManager manager) {
+    return DefaultMetricsSystem.instance().register(METRICS_SOURCE_NAME,
+        "SCM Replication manager (closed container replication) related "
+            + "metrics",
+        new ReplicationManagerMetrics(manager));
```

Comment:
thanks for the work! NIT: if RM  start->stop->start, `ReplicationManagerMetrics#create` will be call twice ,and thus `new ReplicationManagerMetrics(manager)` will be called twice, it seems a little confuse to create the `ReplicationManagerMetrics ` twice. i think it is better off  using `register` instead of `create` for `ReplicationManagerMetrics`, and making ReplicationManagerMetrics singleton

Reply:
Oh, thanks for the comments, it seems to be reasonable to adopt singleton pattern at first glance.
But actually there are things to be made clear:
1. When does start->stop->start happen? and the purpose ?
2. If we adopt singleton, then there's a behavior change:
    - With singleton, we have only one metrics object, and after restart RM(without restarting the daemon of course) will start counting from an old metrics base.
    - Without singleton, RM start with a freshly created object all init to 0.
    So what is the desired behavior?

Reply:
In general I prefer to avoid singletons, as they can make testing difficult, especially around Mini-Cluster tests.

I think its reasonable for a RM restart to reset the counts to zero, as it is a new instance. A RM restart should only happen in tests, as to restart it in a real cluster, you must restart all of SCM.